### ISSUE TITLE
Systemd: Failed to start User Manager for UID 10000 with NIS configuration

### ISSUE THREAD

**Issue by keyslapper** [systemd/systemd#12702]
**systemd version the issue has been seen with**
> $ systemctl --version
> systemd 241 (v241-8.git9ef65cb.fc30)
> +PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > Fedora Workstation 30

**Expected behaviour you didn't see**
 > systemctl status should show "State:  running"

**Unexpected behaviour you saw**
 > $ systemctl status
> ● FC30VBox
>     State: degraded
>      Jobs: 0 queued
>    Failed: 1 units
>     Since: Thu 2019-05-30 09:37:19 EDT; 7min ago

> $ systemctl list-units --failed
>   UNIT               LOAD   ACTIVE SUB    DESCRIPTION
> ● user@10000.service loaded failed failed User Manager for UID 10000
> 
> $ systemctl status user@10000.service
> ● user@10000.service - User Manager for UID 10000
>    Loaded: loaded (/usr/lib/systemd/system/user@.service; static; vendor preset: disabled)
>    Active: failed (Result: protocol) since Fri 2019-05-24 13:29:58 EDT; 22min ago
>      Docs: man:user@.service(5)
>   Process: 7080 ExecStart=/usr/lib/systemd/systemd --user (code=exited, status=217/USER)
>  Main PID: 7080 (code=exited, status=217/USER)
> 
> May 24 13:29:58 FC30VBox systemd[1]: Starting User Manager for UID 10000...
> May 24 13:29:58 FC30VBox systemd[7080]: user@10000.service: Failed to determine user credentials: No such process
> May 24 13:29:58 FC30VBox systemd[7080]: user@10000.service: Failed at step USER spawning /usr/lib/systemd/systemd: No such process
> May 24 13:29:58 FC30VBox systemd[1]: user@10000.service: Failed with result 'protocol'.
> May 24 13:29:58 FC30VBox systemd[1]: Failed to start User Manager for UID 10000.

And the relevant section from journalctl -xe:

>  A start job for unit user@10000.service has begun execution.
>  
>  The job identifier is 3006.
> May 24 14:05:37 FC30VBox systemd[16465]: user@10000.service: Failed to determine user credentials: No such process
> May 24 14:05:37 FC30VBox systemd[16465]: user@10000.service: Failed at step USER spawning /usr/lib/systemd/systemd: No such process
>  Subject: Process /usr/lib/systemd/systemd could not be executed
>  Defined-By: systemd
>  Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
> 
>  The process /usr/lib/systemd/systemd could not be executed and failed.
> 
>  The error number returned by this process is ERRNO.
> May 24 14:05:37 FC30VBox systemd[1]: user@10000.service: Failed with result 'protocol'.
>  Subject: Unit failed
>  Defined-By: systemd
>  Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
> 
>  The unit user@10000.service has entered the 'failed' state with result 'protocol'.
> May 24 14:05:37 FC30VBox systemd[1]: Failed to start User Manager for UID 10000.
>  Subject: A start job for unit user@10000.service has failed
>  Defined-By: systemd
>  Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
> 
>  A start job for unit user@10000.service has finished with a failure.

**Steps to reproduce the problem**
Install Fedora Workstation 30 in VirtualBox VM
Update all packages
Install libnsl, nss_nis autofs, ypbind, yp-tools
Configure autofs and ypbind appropriately
$ authselect select nis
Verify ypwhich and automount work:

> $ authselect current
> Profile ID: nis
> Enabled features: None

Calls to ypwhich, ypcat, domainname, etc. all display the expected NIS domain. I am able to log in, but constantly have issues with the system freezing for a moment or two. This user@xxx.service service is the only one that always fails.

If not for the fact I'm using an identical configuration from a Fedora 29 VM that works perfectly running systemd 239, I'd absolutely assume I was doing something wrong. As it is, either the bug mentioned in [this thread](https://github.com/systemd/systemd/issues/9656) (or some variant of it) is back, or I've missed something _new_ in moving to FC30.

Just an FYI: The UID shown in the output above has been changed, but since it's an NIS user and not a local user, 1000 would be wrong.  Our NIS service has user accounts that run from around 10000 to over 1000000.

Any suggestions would be appreciated.
Cheers!
Lou

**Comment by yuwata** [comment]
Please show `systemctl cat systemd-logind.service`.

**Comment by keyslapper** [comment]
Thank you for taking a look
Here's the full output from the command you gave:

`# /usr/lib/systemd/system/systemd-logind.service`
`#  SPDX-License-Identifier: LGPL-2.1+`
`#`
`#  This file is part of systemd.`
`#`
`#  systemd is free software; you can redistribute it and/or modify it`
`#  under the terms of the GNU Lesser General Public License as published by`
`#  the Free Software Foundation; either version 2.1 of the License, or`
`#  (at your option) any later version.`

`[Unit]`
`Description=Login Service`
`Documentation=man:systemd-logind.service(8) man:logind.conf(5)`
`Documentation=https://www.freedesktop.org/wiki/Software/systemd/logind`
`Documentation=https://www.freedesktop.org/wiki/Software/systemd/multiseat`
`Wants=user.slice`
`After=nss-user-lookup.target user.slice`

`# Ask for the dbus socket.`
`Wants=dbus.socket`
`After=dbus.socket`

`[Service]`
`BusName=org.freedesktop.login1`
`CapabilityBoundingSet=CAP_SYS_ADMIN CAP_MAC_ADMIN CAP_AUDIT_CONTROL CAP_CHOWN CAP_KILL CAP_DAC_READ_SEARCH CAP_DAC_OVERRIDE CAP_FOWNER CAP_SYS_TTY_CONFIG`
`ExecStart=/usr/lib/systemd/systemd-logind`
`FileDescriptorStoreMax=512`
`#IPAddressDeny=any`
`LockPersonality=yes`
`MemoryDenyWriteExecute=yes`
`Restart=always`
`RestartSec=0`
`RestrictAddressFamilies=AF_UNIX AF_NETLINK`
`RestrictNamespaces=yes`
`RestrictRealtime=yes`
`RestrictSUIDSGID=yes`
`SystemCallArchitectures=native`
`SystemCallErrorNumber=EPERM`
`SystemCallFilter=@system-service`
`WatchdogSec=3min`

`# Increase the default a bit in order to allow many simultaneous logins since`
`# we keep one fd open per session.`
`LimitNOFILE=524288`

`# /usr/lib/systemd/system/systemd-logind.service.d/nss_nis.conf`
`[Service]`
`IPAddressAllow=any`
`RestrictAddressFamilies=AF_UNIX AF_NETLINK AF_INET AF_INET6`

As far as I can tell, this is correct, but you're the expert.

Thank you again.
Lou

**Comment by keyslapper** [comment]
Not to be impatient, but is there anything I can try, or any other info I can provide here to help move this issue forward?

**Comment by RobbieTheK** [comment]
Almost exact same here except Fedora 29. I commented at the [possible related closed bug](https://github.com/systemd/systemd/issues/9656#issuecomment-502112924) and uploaded a coredump.

**Comment by keyslapper** [comment]
@RobbieTheK I don't think the `IPAddressDeny=` setting will fix anything for you - I have the same config in my FC29 VM and it works fine.  I believe that's synonymous with `IPAddressAllow=any`
I think the LimitNOFILE config is the only thing that's different between my FC29 and FC30 VMs.

I still can't find anything that suggests either a cause or a workaround.

**Comment by poettering** [comment]
This suggests UID 10000 is simply not resolvable at the moment systemd forks off user@10000.service. Are you sure NIS is properly set up and connect to its server? Does resolving that UID work with "getent passwd 10000" at the moment you log in? Do you have user "lingering" enabled? Is NIS properly packaged and orders its services before nss-user-lookup.target and pulls it in? Does it properly pull in network-online.target and orders itself after that?

This is really something you have to talk with the NIS maintainers of your distro about, we don't really know much about NIS upstream (we don't live in 1992 anymore ;-))...

Anyway, the hooks are all there to order NIS stuff properly, it's just a matter of doing so,

**Comment by keyslapper** [comment]
Well, it's a bit slow, but 'getent passwd 10000' does return my userid.
RE: user lingering, I haven't touched it, and I'm not sure why it would matter - this is a VM that is only used with the one account.  I only log out when the VM is shut down, so having the user manager "linger" shouldn't matter.  The problem is that the user manager crashes while I'm logged in.
As far as NIS configuration, yes I believe the configuration is correct - it's identical to my FC29 VM, which works perfectly.  Both are configured to the same yp domain ('ypwhich' returns the same value) and both are able to resolve addresses on the same subnets.

I'm not sure how to go about verifying the NIS packaging and ordering of services.

Thanks for the input.

**Comment by poettering** [comment]
what does "is a bit slow" mean? How can it be "slow"?

user@10000.service gets started when the user 10000 logs in first. When is that? is that a real "human" user? or some cronjob?

**Comment by RobbieTheK** [comment]
> This suggests UID 10000 is simply not resolvable at the moment systemd forks off [user@10000.service](mailto:user@10000.service). Are you sure NIS is properly set up and connect to its server? Does resolving that UID work with "getent passwd 10000" at the moment you log in? Do you have user "lingering" enabled? Is NIS properly packaged and orders its services before nss-user-lookup.target and pulls it in? Does it properly pull in network-online.target and orders itself after that?
> 
> This is really something you have to talk with the NIS maintainers of your distro about, we don't really know much about NIS upstream (we don't live in 1992 anymore ;-))...
> 
> Anyway, the hooks are all there to order NIS stuff properly, it's just a matter of doing so,

Apologies for the confusion. In the [other thread](https://github.com/systemd/systemd/issues/9656#issuecomment-502112924),  it's not user 10,000, it's user 0:
`Failed to start User Manager for UID 0.`

```
getent passwd 0
root:x:0:0:root:/root:/bin/bash
```

**Comment by keyslapper** [comment]
@poettering 
By "slow" I mean when I run "getent passwd 10000", it takes around 3 seconds to return the value (and it does return the correct string).  My FC29 VM returns immediately.  These are both VirtualBox VMs running as development systems - VM setup is identical on both, I've been doing this on the same network since around FC18.   My laptop is running Windows, but I do all my "real" work in a Linux VM connected to a network filesystem via NIS login.

So to answer your other questions, this is a real user, logging into a Gnome shell on a Fedora VM.  All the details in my OP were obtained from this login.  I am not being kicked off the system, but I see a lot of short-lived hangs - more than enough that it's not worth moving from FC29.


Just to give a little background, I started having trouble when Fedora switched to authselect (FC28), and there was no NIS profile included.  As far as I can tell, that has been resolved, and FC29 works quite well with the standard authselect setup - I didn't have to do much more than "authselect select nis" and set my domain in /etc/yp.conf.  I have to admit I was really annoyed with the decision to adopt authselect until I saw this in action.

When I moved to FC30, I used the same set of commands and configuration changes that worked with FC29, but I am continually seeing the system freeze for 30 seconds or more (with all kinds of activities and applications), and the only thing I can find wrong is that User@10000.service is in a failed state.  The logs ("journalctl -xe" and "systemctl status User@10000.service") show that it is being started when I log in, but they immediately show it entering a failed state.  I have found almost nothing on Google to indicate anyone else has ever seen this problem in conjunction with NIS, but what little I did see points to systemd.  That's what led me here.  I don't know what to try next, so hopefully someone here with enough insight to this corner of systemd can help.

Thank you again

**Comment by poettering** [comment]
@RobbieTheK you appear to have a very different issue

**Comment by poettering** [comment]
@keyslapper my educated guess is that your NIS lookup is just very unreliable, and sometimes it fails, when user@10000.service tries to resolve the username.

When the user fails to resolve and thus user@10000.service fails, can you then start it manually (i.e. does "systemctl user@100000.service" succeed?

If NIS is just flaky, then please contact the NIS maintainers.

The error you see regarding "Failed to determine user credentials: No such process" and "Failed at step USER spawning /usr/lib/systemd/systemd: No such process" just means that your NIS user is not resolvable when user@10000.service is about to start, and hence systemd can't start it. How that happened is really in NIS territory, you'd have to find that yourself with help of the NIS maintainers I fear...

**Comment by RobbieTheK** [comment]
> @RobbieTheK you appear to have a very different issue

Do you think the [other thread](https://github.com/systemd/systemd/issues/9656#issuecomment-502112924) is more appropriate or open a new issue?

**Comment by keyslapper** [comment]
@poettering 
For awhile, I thought exactly the same thing, but I suspect I'd see the same behavior on any host if that were so.  And I have tried restarting the user management daemon - it always fails immediately with the same errors.
I wonder if I'm not quite understanding what you mean by "your NIS user is not resolvable".  If I understand right, there are several possibilities:
The user is invalid on the NIS server (not the case, or I wouldn't be able to log in at all)
The NIS server is not resolvable (probably would also prevent login)
The NIS server is "flaky" meaning sometimes it works, sometimes it doesn't (I suspect this would affect most systems connecting to the server, but it doesn't)

What confuses me more is that if the user management service should be able to start when the NIS user can be resolved, why would it fail after I get the expected result from the getent command above?
Is it possible the user manager is just timing out too quickly?

I'll try to find the NIS maintainers for my distro.

Thank you again.

**Comment by poettering** [comment]
when systemd resolves the user name it ultimately just calls glibc's getpwnam(), and that appears to fail. I am not sure why. You might be able to figure out more by doing `strace -f -p 1 -s 500 -o log` before starting user@10000.service manually, then having a look at the trace this generated. The lookup happens in the child that is forked off from PID 1 and is supposed to become the "systemd --user" instance. Maybe you can see from the traffic the NIS nss module generates already what is going on there. If not, please paste

**Comment by keyslapper** [comment]
BTW, this is what I see when I try to start the user manager:
```
$ systemctl start user@10000.service
Job for user@10000.service failed because the service did not take the steps required by its unit configuration.
See "systemctl status user@10000.service" and "journalctl -xe" for details.
```

And the journalctl -xe output:
```
$ journalctl -xe
-- Subject: A start job for unit user@10000.service has begun execution
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit user@10000.service has begun execution.
--
-- The job identifier is 2837.
Jun 18 11:12:08 FC30VBox systemd[2887]: user@10000.service: Failed to determine user credentials: No such process
Jun 18 11:12:08 FC30VBox systemd[2887]: user@10000.service: Failed at step USER spawning /usr/lib/systemd/systemd: No such process
-- Subject: Process /usr/lib/systemd/systemd could not be executed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- The process /usr/lib/systemd/systemd could not be executed and failed.
--
-- The error number returned by this process is ERRNO.
Jun 18 11:12:08 FC30VBox systemd[1]: user@10000.service: Failed with result 'protocol'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- The unit user@10000.service has entered the 'failed' state with result 'protocol'.
Jun 18 11:12:08 FC30VBox systemd[1]: Failed to start User Manager for UID 10000.
-- Subject: A start job for unit user@10000.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- A start job for unit user@10000.service has finished with a failure.
--
-- The job identifier is 2837 and the job result is failed.
```

And systemctl status:
```
$ systemctl status user@10000.service
● user@10000.service - User Manager for UID 10000
   Loaded: loaded (/usr/lib/systemd/system/user@.service; static; vendor preset: disabled)
   Active: failed (Result: protocol) since Tue 2019-06-18 11:12:08 EDT; 2min 23s ago
     Docs: man:user@.service(5)
  Process: 2887 ExecStart=/usr/lib/systemd/systemd --user (code=exited, status=217/USER)
 Main PID: 2887 (code=exited, status=217/USER)

Jun 18 11:12:08 FC30VBox systemd[1]: Starting User Manager for UID 10000...
Jun 18 11:12:08 FC30VBox systemd[2887]: user@10000.service: Failed to determine user credentials: No such process
Jun 18 11:12:08 FC30VBox systemd[2887]: user@10000.service: Failed at step USER spawning /usr/lib/systemd/systemd: No such process
Jun 18 11:12:08 FC30VBox systemd[1]: user@10000.service: Failed with result 'protocol'.
Jun 18 11:12:08 FC30VBox systemd[1]: Failed to start User Manager for UID 10000.

```

There are almost 2000 lines of very confusing text in the strace log.  Aside from that, it contains some details I've thus far kept obfuscated (i.e. the actual NIS domain, and my actual UID is not 10000).  What should I look for?

Again, your help is much appreciated.

**Comment by poettering** [comment]
if you are fine we me having a private look at the strace stuff, send it to me personally via mail (lennart at poettering dot net). Otherwise you have to analyze it yourself, but I am not sure what to recommend to you, you need to know what you are doing to read an strace I figure. Maybe look for failed connect()s or so

**Comment by keyslapper** [comment]
I really appreciate the offer, and if it were my personal setup I'd have no problem with that, but I'm afraid it's a corporate network, and I'm not at liberty to disclose details I don't understand.
I did find a half dozen attempts to connect th /var/run/nscd/socket that failed, but I don't think that's the problem - all subsequent connect attempts succeed.
There are 8 successful connections to our name server (port 53, dns lookups).

After that, there are 4 connections to localhost (127.0.0.1, port 111, rpcbind) - all successful.

Then there are 2 connections to /run/systemd/journal/socket to report the errors I've already included.

There are no references of any kind to the NIS server, either by name or by IP.

I also see the following, but I'm not sure if this is innocuous or problematic:

```
12542 openat(AT_FDCWD, "/lib64/libnss_systemd.so.2", O_RDONLY|O_CLOEXEC) = 3
12542 read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0j\0\0\0\0\0\0@\0\0\0\0\0\0\0X\376\4\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\36\0\35\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0 ^\0\0\0\0\0\0 ^\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0`\0\0\0\0\0\0\0`\0\0\0\0\0\0\0`\0\0\0\0\0\0A,\3\0\0\0\0\0A,\3\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0\220\3\0\0\0\0\0\0\220\3\0\0\0\0\0\0\220\3\0\0\0\0\08\27\1\0\0\0\0\08\27\1\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0\10\252\4\0\0\0\0\0\10\272\4\0\0\0\0\0\10\272\4\0\0\0\0\0\330-\0\0\0\0\0\0\3606\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0`\310\4\0\0\0\0\0`\330\4\0\0\0\0\0`\330\4\0\0\0\0\0@\2\0\0\0\0\0\0@\2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0$\0\0\0\0\0\0\0$\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\7\0\0\0\4\0\0\0\10\252\4\0\0\0\0\0\10\272\4\0\0\0\0\0\10\272\4\0\0\0\0\0\4\0\0\0\0\0\0\0h\0\0\0\0\0\0\0\10\0\0\0\0\0\0\0P\345td\4\0\0\0\20'\4\0\0\0\0\0\20'\4\0\0\0\0\0\20'\4\0\0\0\0\0\4\17\0\0\0\0\0\0\4\17\0\0"..., 832) = 832
```

The first error written to the journal is shortly after that message.

Thank you

**Comment by poettering** [comment]
do you see libnss_nis even opened? is nis listed in /etc/nsswitch.conf?

**Comment by poettering** [comment]
in which range is your uid anyway? See https://systemd.io/UIDS-GIDS.html#summary

**Comment by keyslapper** [comment]
My UID is between 74000 and 75000 - roughly halfway.  Has been for almost 12 years, and has always been on the same NIS server.

The openat message I gave above was the last one before the failure, there are other libnss_* libraries opened, in the following order:

```
12542 openat(AT_FDCWD, "/lib64/libnss_files.so.2", O_RDONLY|O_CLOEXEC) = 3
12542 read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\3605\0\0\0\0\0\0@\0\0\0\0\0\0\0P#\2\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\"\0!\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\240!\0\0\0\0\0\0\240!\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0000\0\0\0\0\0\0\0000\0\0\0\0\0\0\0000\0\0\0\0\0\0yi\0\0\0\0\0\0yi\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0\240\0\0\0\0\0\0\0\240\0\0\0\0\0\0\0\240\0\0\0\0\0\0\310\37\0\0\0\0\0\0\310\37\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0\200\315\0\0\0\0\0\0\200\335\0\0\0\0\0\0\200\335\0\0\0\0\0\0\10\4\0\0\0\0\0\0\330i\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0\230\315\0\0\0\0\0\0\230\335\0\0\0\0\0\0\230\335\0\0\0\0\0\0 \2\0\0\0\0\0\0 \2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0D\0\0\0\0\0\0\0D\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0\270\2\0\0\0\0\0\0\270\2\0\0\0\0\0\0\270\2\0\0\0\0\0\0 \0\0\0\0\0\0\0 \0\0\0\0\0\0\0\10\0\0\0\0\0\0\0P\345td\4\0\0\0\0\242\0\0\0\0\0\0\0\242\0\0\0\0\0\0\0\242\0\0\0\0\0\0\f\3\0\0\0\0\0\0\f\3\0\0"..., 832) = 832
12542 fstat(3, {st_mode=S_IFREG|0755, st_size=142288, ...}) = 0
12542 mmap(NULL, 83800, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f66cfcbf000
12542 mprotect(0x7f66cfcc2000, 40960, PROT_NONE) = 0
12542 mmap(0x7f66cfcc2000, 28672, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3000) = 0x7f66cfcc2000
12542 mmap(0x7f66cfcc9000, 8192, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xa000) = 0x7f66cfcc9000
12542 mmap(0x7f66cfccc000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f66cfccc000
12542 mmap(0x7f66cfcce000, 22360, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f66cfcce000
12542 close(3)                          = 0
. . .
12542 openat(AT_FDCWD, "/lib64/libnss_nis.so.2", O_RDONLY|O_CLOEXEC) = 3
12542 read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0p%\0\0\0\0\0\0@\0\0\0\0\0\0\0\270@\1\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\35\0\34\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0p\33\0\0\0\0\0\0p\33\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0 \0\0\0\0\0\0\0 \0\0\0\0\0\0\0 \0\0\0\0\0\0I\212\0\0\0\0\0\0I\212\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0\260\0\0\0\0\0\0\0\260\0\0\0\0\0\0\0\260\0\0\0\0\0\0\240\32\0\0\0\0\0\0\240\32\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0000\314\0\0\0\0\0\0000\334\0\0\0\0\0\0000\334\0\0\0\0\0\0\342\3\0\0\0\0\0\0\0\10\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0H\314\0\0\0\0\0\0H\334\0\0\0\0\0\0H\334\0\0\0\0\0\0000\2\0\0\0\0\0\0000\2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0$\0\0\0\0\0\0\0$\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0\200\312\0\0\0\0\0\0\200\312\0\0\0\0\0\0\200\312\0\0\0\0\0\0 \0\0\0\0\0\0\0 \0\0\0\0\0\0\0\10\0\0\0\0\0\0\0P\345td\4\0\0\0\224\264\0\0\0\0\0\0\224\264\0\0\0\0\0\0\224\264\0\0\0\0\0\0\214\2\0\0\0\0\0\0\214\2\0\0"..., 832) = 832
12542 lseek(3, 51840, SEEK_SET)         = 51840
12542 read(3, "\4\0\0\0\20\0\0\0\5\0\0\0GNU\0\2\0\0\300\4\0\0\0\3\0\0\0\0\0\0\0", 32) = 32
12542 fstat(3, {st_mode=S_IFREG|0755, st_size=83960, ...}) = 0
12542 lseek(3, 51840, SEEK_SET)         = 51840
12542 read(3, "\4\0\0\0\20\0\0\0\5\0\0\0GNU\0\2\0\0\300\4\0\0\0\3\0\0\0\0\0\0\0", 32) = 32
12542 mmap(NULL, 58416, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f66d0e90000
12542 mmap(0x7f66d0e92000, 36864, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f66d0e92000
12542 mmap(0x7f66d0e9b000, 8192, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xb000) = 0x7f66d0e9b000
12542 mmap(0x7f66d0e9d000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xc000) = 0x7f66d0e9d000
12542 close(3)                          = 0
. . .
12542 openat(AT_FDCWD, "/lib64/libnss_dns.so.2", O_RDONLY|O_CLOEXEC) = 5
12542 read(5, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\23\0\0\0\0\0\0@\0\0\0\0\0\0\0\370\244\0\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\"\0!\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\240\16\0\0\0\0\0\0\240\16\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0\20\0\0\0\0\0\0\0\20\0\0\0\0\0\0\0\20\0\0\0\0\0\0\3417\0\0\0\0\0\0\3417\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0P\0\0\0\0\0\0\0P\0\0\0\0\0\0\0P\0\0\0\0\0\0X\7\0\0\0\0\0\0X\7\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0p]\0\0\0\0\0\0pm\0\0\0\0\0\0pm\0\0\0\0\0\0p\3\0\0\0\0\0\0x\3\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0\210]\0\0\0\0\0\0\210m\0\0\0\0\0\0\210m\0\0\0\0\0\0000\2\0\0\0\0\0\0000\2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0D\0\0\0\0\0\0\0D\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0\270\2\0\0\0\0\0\0\270\2\0\0\0\0\0\0\270\2\0\0\0\0\0\0 \0\0\0\0\0\0\0 \0\0\0\0\0\0\0\10\0\0\0\0\0\0\0P\345td\4\0\0\0\244P\0\0\0\0\0\0\244P\0\0\0\0\0\0\244P\0\0\0\0\0\0\214\0\0\0\0\0\0\0\214\0\0\0"..., 832) = 832
12542 fstat(5, {st_mode=S_IFREG|0755, st_size=44408, ...}) = 0
12542 mmap(NULL, 28904, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 5, 0) = 0x7f66cfbbc000
12542 mmap(0x7f66cfbbd000, 16384, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0x1000) = 0x7f66cfbbd000
12542 mmap(0x7f66cfbc1000, 4096, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0x5000) = 0x7f66cfbc1000
12542 mmap(0x7f66cfbc2000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0x5000) = 0x7f66cfbc2000
12542 close(5)                          = 0
. . .
12542 openat(AT_FDCWD, "/lib64/libnss_myhostname.so.2", O_RDONLY|O_CLOEXEC) = 5
12542 read(5, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P4\0\0\0\0\0\0@\0\0\0\0\0\0\0(j\1\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\34\0\33\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\330!\0\0\0\0\0\0\330!\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0000\0\0\0\0\0\0\0000\0\0\0\0\0\0\0000\0\0\0\0\0\0\361\243\0\0\0\0\0\0\361\243\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0\340\0\0\0\0\0\0\0\340\0\0\0\0\0\0\0\340\0\0\0\0\0\0{\\\0\0\0\0\0\0{\\\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0\260=\1\0\0\0\0\0\260M\1\0\0\0\0\0\260M\1\0\0\0\0\0\330\"\0\0\0\0\0\0x+\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0@[\1\0\0\0\0\0@k\1\0\0\0\0\0@k\1\0\0\0\0\0@\2\0\0\0\0\0\0@\2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0$\0\0\0\0\0\0\0$\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\7\0\0\0\4\0\0\0\260=\1\0\0\0\0\0\260M\1\0\0\0\0\0\260M\1\0\0\0\0\0\0\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0P\345td\4\0\0\0\370\34\1\0\0\0\0\0\370\34\1\0\0\0\0\0\370\34\1\0\0\0\0\0\254\3\0\0\0\0\0\0\254\3\0\0"..., 832) = 832
12542 fstat(5, {st_mode=S_IFREG|0755, st_size=94504, ...}) = 0
12542 mmap(NULL, 96552, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 5, 0) = 0x7f66cfba4000
12542 mmap(0x7f66cfba7000, 45056, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0x3000) = 0x7f66cfba7000
12542 mmap(0x7f66cfbb2000, 24576, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0xe000) = 0x7f66cfbb2000
12542 mmap(0x7f66cfbb8000, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 5, 0x13000) = 0x7f66cfbb8000
12542 close(5)                          = 0
. . .
12542 openat(AT_FDCWD, "/lib64/libnss_systemd.so.2", O_RDONLY|O_CLOEXEC) = 3
12542 read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0j\0\0\0\0\0\0@\0\0\0\0\0\0\0X\376\4\0\0\0\0\0\0\0\0\0@\08\0\n\0@\0\36\0\35\0\1\0\0\0\4\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0 ^\0\0\0\0\0\0 ^\0\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\5\0\0\0\0`\0\0\0\0\0\0\0`\0\0\0\0\0\0\0`\0\0\0\0\0\0A,\3\0\0\0\0\0A,\3\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\4\0\0\0\0\220\3\0\0\0\0\0\0\220\3\0\0\0\0\0\0\220\3\0\0\0\0\08\27\1\0\0\0\0\08\27\1\0\0\0\0\0\0\20\0\0\0\0\0\0\1\0\0\0\6\0\0\0\10\252\4\0\0\0\0\0\10\272\4\0\0\0\0\0\10\272\4\0\0\0\0\0\330-\0\0\0\0\0\0\3606\0\0\0\0\0\0\0\20\0\0\0\0\0\0\2\0\0\0\6\0\0\0`\310\4\0\0\0\0\0`\330\4\0\0\0\0\0`\330\4\0\0\0\0\0@\2\0\0\0\0\0\0@\2\0\0\0\0\0\0\10\0\0\0\0\0\0\0\4\0\0\0\4\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0p\2\0\0\0\0\0\0$\0\0\0\0\0\0\0$\0\0\0\0\0\0\0\4\0\0\0\0\0\0\0\7\0\0\0\4\0\0\0\10\252\4\0\0\0\0\0\10\272\4\0\0\0\0\0\10\272\4\0\0\0\0\0\4\0\0\0\0\0\0\0h\0\0\0\0\0\0\0\10\0\0\0\0\0\0\0P\345td\4\0\0\0\20'\4\0\0\0\0\0\20'\4\0\0\0\0\0\20'\4\0\0\0\0\0\4\17\0\0\0\0\0\0\4\17\0\0"..., 832) = 832
12542 fstat(3, {st_mode=S_IFREG|0755, st_size=329176, ...}) = 0
12542 mmap(NULL, 323832, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f66cfb54000
12542 mmap(0x7f66cfb5a000, 208896, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0x7f66cfb5a000
12542 mmap(0x7f66cfb8d000, 73728, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x39000) = 0x7f66cfb8d000
12542 mmap(0x7f66cfb9f000, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x4a000) = 0x7f66cfb9f000
12542 mmap(0x7f66cfba3000, 248, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f66cfba3000
12542 close(3)                          = 0
```
All of the files in question are present in /lib64/.

My nsswitch.conf file does mention nis - it's using the default NIS configuration generated by authselect:
```
aliases:    files nis
automount:  files nis
ethers:     files nis
group:      files nis systemd
hosts:      files nis dns myhostname
initgroups: files nis
netgroup:   files nis
networks:   files nis
passwd:     files nis systemd
protocols:  files nis
publickey:  files nis
rpc:        files nis
services:   files nis
shadow:     files nis
```

That's actually the entire thing minus comments - I don't have user modifications to the nsswitch.conf.

I'm pretty sure libnss_nis does get opened at some point, else I wouldn't even be able to log in, would I?

Thank you

**Comment by poettering** [comment]
btw, did you try to turn off selinux?

so nss-nis gets loaded, which means you now need to find where it does what it does, and what happens there. I don't know NIS and without the trace i cannot even guess, so you have to figure that out on your own, or with help of the NIS maintainers... not sure this is actionable for us anymore at this point.

**Comment by keyslapper** [comment]
I've had selinux in permissive mode, but just for yuks, I disabled it and rebooted.  No change.  And it looks like something else is happening now, because I can't even log in.

I appreciate your time here, but I'm clearly going to have to put this to the Fedora maintainers.

We can leave this open if you like, and I'll comment here if the problem is ever resolved.

**Comment by RobbieTheK** [comment]
I'm seeing this on various workstations and servers in our NIS network:
```
● user@1201.service            loaded failed failed User Manager for UID 1201
● user@6105.service            loaded failed failed User Manager for UID 6105
```
And with UID 0 as well. I [created a bug report](https://bugzilla.redhat.com/show_bug.cgi?id=1721574) which contains a strace. I'm happy to provide more. I do see logs such as below:

```
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.a11y.Bus' requested by ':1.40' (uid=1201 pid=7073 comm="gnome-shell --sm-client-id 10ee976375cc69c95715577")
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Successfully activated service 'org.a11y.Bus'
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: Policy to allow eavesdropping in /usr/share/defaults/at-spi2/accessibility.conf +15: Eavesdropping is deprecated and ignored
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: Policy to allow eavesdropping in /usr/share/defaults/at-spi2/accessibility.conf +17: Eavesdropping is deprecated and ignored
Jun 17 09:18:51 ourworkstation journal[7092]: Ready
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.systemd1' requested by ':1.42' (uid=1201 pid=7092 comm="/usr/bin/dbus-broker-launch --config-file=/usr/sha")
Jun 17 09:18:51 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Jun 17 09:18:51 ourworkstation dbus-broker-launch[7092]: Activation request for 'org.a11y.atspi.Registry' failed.
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.portal.IBus' requested by ':1.43' (uid=1201 pid=7104 comm="ibus-daemon --xim --panel disable ")
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.systemd1' requested by ':1.42' (uid=1201 pid=7092 comm="/usr/bin/dbus-broker-launch --config-file=/usr/sha")
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Jun 17 09:18:55 ourworkstation dbus-broker-launch[7092]: Activation request for 'org.a11y.atspi.Registry' failed.
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.systemd1' requested by ':1.42' (uid=1201 pid=7092 comm="/usr/bin/dbus-broker-launch --config-file=/usr/sha")
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Jun 17 09:18:55 ourworkstation dbus-broker-launch[7092]: Activation request for 'org.a11y.atspi.Registry' failed.
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.systemd1' requested by ':1.42' (uid=1201 pid=7092 comm="/usr/bin/dbus-broker-launch --config-file=/usr/sha")
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Jun 17 09:18:55 ourworkstation dbus-broker-launch[7092]: Activation request for 'org.a11y.atspi.Registry' failed.
Jun 17 09:18:55 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Successfully activated service 'org.freedesktop.portal.IBus'
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.freedesktop.impl.portal.PermissionStore' requested by ':1.39' (uid=1201 pid=7073 comm="gnome-shell --sm-client-id 10ee976375cc69c95715577")
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.gnome.Shell.CalendarServer' requested by ':1.39' (uid=1201 pid=7073 comm="gnome-shell --sm-client-id 10ee976375cc69c95715577")
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: The XKEYBOARD keymap compiler (xkbcomp) reports:
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: > Internal error:   Could not resolve keysym XF86MonBrightnessCycle
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: > Internal error:   Could not resolve keysym XF86RotationLockToggle
Jun 17 09:18:57 ourworkstation /usr/libexec/gdm-x-session[6755]: Errors from xkbcomp are not fatal to the X server
Jun 17 09:18:58 ourworkstation /usr/libexec/gdm-x-session[6755]: dbus-daemon[6771]: [session uid=1201 pid=6771] Activating service name='org.gnome.evolution.dataserver.Sources5' requested by ':1.50' (uid=1201 pid=7148 comm="/usr/libexec/gnome-shell-calendar-server ")
Jun 17 09:18:58 ourworkstation journal[7073]: g_dir_open_with_errno: assertion 'path != NULL' failed
Jun 17 09:18:58 ourworkstation journal[7073]: g_filename_to_utf8: assertion 'opsysstring != NULL' failed
Jun 17 09:18:58 ourworkstation journal[7073]: g_dir_open_with_errno: assertion 'path != NULL' failed
Jun 17 09:18:58 ourworkstation journal[7073]: g_filename_to_utf8: assertion 'opsysstring != NULL' failed
```

**Comment by keyslapper** [comment]
I don't know if it makes any difference, but after I lost gnome-shell login altogether, I completely reset my authselect config with 'authselect select nis --force' to make sure I was discarding anything I'd added to the configuration.  No change logging in.

On a hunch, I switched to a different tty terminal (Ctrl-Alt-F4) and I was able to log in at the command line with no problem.  The user service still shows up as failed, and logging in on a tty terminal doesn't actually help me, but I think it shows that my NIS configuration is correct at least.

I am still getting the systemd errors:
```
user@xxxxx.service Failed to determine user credentials: No such process
user@xxxxx.service: Failed at step USER spawning /usr/lib/systemd/systemd: No such process
user@xxxxx.service: filed with result 'protocol'.
Filed to start User Manager for UID xxxxx
```

I wasn't sure if this was somehow related to the fact that gnome-shell is still pretty buggy in this release, so I did the above after rebooting and without logging in to gnome-shell at all.

Still trying to figure out where to post my issue with Fedora ...

**Comment by mborysow** [comment]
Is this related to the issues here:
https://bugzilla.redhat.com/show_bug.cgi?id=1705641
https://bugzilla.redhat.com/show_bug.cgi?id=1716382

**Comment by keyslapper** [comment]
Fairly certain it's not related to these, but who knows.

I've had very little time to spend on this lately, but I've updated my FC30 VM a number of times and retried this, but I'm still getting the same problems.  Always failure of the user@<uid> service.  Usually even get a segfault in gnome-shell.

I've also verified that my configuration is identical to the one that works perfectly in FC29.  I can't tell what it is, but something critical to NIS functionality is broken in FC30.

I should also note that my personal laptop is currently running FC30 (default configuration) and I have no issues.

**Comment by keszybz** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1721574 seems to be the appropriate place to track this. The issue seem to be specific to nss_nis, and probably depend on the pam setup and nis version and other details of configuration, so this should first be diagnosed at the distro level.

**Comment by keyslapper** [comment]
@keszybz, I respectfully disagree with your conclusion here.  As I've mentioned many times, I am not having trouble with FC29, only FC30.  The bugzilla link you posted applies to FC29, and is clearly not the issue I'm having.

**Comment by keszybz** [comment]
@keyslapper it is still a "downstream" issue. The bug depends on pam configuration and other details of setup, so it needs to be resolved in Fedora, not here.

**Comment by keyslapper** [comment]
I'm using exactly the same configuration in both places, so I'm not entirely sure about that either.  Regardless, I would think that any process that handles a bad configuration by crashing without any useful info would be considered broken.

**Comment by brasslan** [comment]
I have a workstation at work that I installed Fedora 25 on in 2017.  Connected it to my work NIS domain and it has been working great through a few years of system upgrades from 25 to 26 to 27 to 28 to 29.  Last week I updated to Fedora 30 and now I have the exact same issue as the OP.  Except that I'm on bare metal and I think my system is slower than the VM described above.  I have 64GB of ram in this system and there was a memory leak eating all the ram right after the upgrade.  I turned off some services, like libvirtd and others and changed the run target to multi-user so the GUI no longer runs, and now I no longer have the problems of no RAM available.  But the entire system is still VERY slow.  For example rpm -qa takes over 2 minutes to list all the installed packages.

Maybe not the same issue as the OP because "getent passwd 99441848" for me comes back quick.  But so many things are so slow the system isn't usable in this state.  Trying to update to Fedora 31 fails with timeout errors (All mirrors were tried) during the package download, but the internet is fine.  But I'm trying it multiple times and it gets farther each time.

I found this thread because I'm getting this error for multiple system UID users in /var/log/messages
Feb  5 12:47:30 ws0 systemd[1]: user@48.service: Failed with result 'protocol'.
Feb  5 12:47:31 ws0 systemd[1]: Failed to start User Manager for UID 48.

My plan is to try the Fedora 31 upgrade over and over again to see if I can download all required packages and then see what happens after the upgrade.  That will update my nss_nis from 3.0-9 to 3.1-2 ypbind from 2.5-4 to 2.6.1-1 and my systemd from 241-12 to 243.5-1.

I've never run a Fedora workstation so long and through so many OS upgrades before and I don't want to format/reload.

keyslapper, did you find a solution to your issue above?

If anyone wants more information from my system, let me know.
Thanks!

**Comment by RobbieTheK** [comment]
> If anyone wants more information from my system, let me know.

Have you tried to [enable/start nscd mentioned by me here](https://bugzilla.redhat.com/show_bug.cgi?id=1716382#c24)?

**Comment by brasslan** [comment]
Thank you Robbie for such a fast response!  And sorry for my extremely slow reply!

Installing and starting that service did help a lot.  But it feels like a band-aid and not a real fix.  I have finally finished downloading all files needed for an upgrade from Fedora 30 to Fedora 31, but I haven't dropped the hammer yet because I'm not sure if things will get better or worse when I do that.  So I think I'll play around for a couple days now that the system is usable again and then try the update Wednesday or Thursday and see what happens.

All of our "official" workstations here are running RHEL, and this Fedora system is just one that I like to play around on because I have access to more software packages.
